### PR TITLE
libbass: 2.4.15 -> 2.4.17

### PR DESCRIPTION
**Pull request by imnotpoz**
## Description of changes
Bumps `libbass` version. I also bumped `libbass_fx` and added `libbassmix`, should I post them in separate PRs after this one or put them all here? These packages also don't have a maintainer set, should I add myself as one or find the author and add them?

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/libraries/audio/libbass/default.nix [hunk 1]
```diff
@@ -2,20 +2,22 @@
...
-        i686_linux = "libbass.so";
-        x86_64-linux = "x64/libbass.so";
+        i686_linux = "libs/x86/libbass.so";
+        x86_64-linux = "libs/x86_64/libbass.so";
+        armv7l-linux = "libs/armhf/libbass.so";
+        aarch64-linux = "libs/aarch64/libbass.so";
       };
       urlpath = "bass24-linux.zip";
-      sha256 = "1lmysxfhy727zskavml3ibg5w876ir88923bm17c21s59w5lh7l8";
+      hash = "sha256-bud59LhdqCHUpTg88+gmZZKiuhCP9Zl2dU/A+Ka05P0=";
```
Comment (FliegendeWurst) [inline pkgs/development/libraries/audio/libbass/default.nix:20]: You also need to update the hash below:
```
      hash = "sha256-1g7J1K9PG6B1CW0DDh4tEDrH1qewYUYgvWBpkScDL2U=";
    };

    bass_fx = {
      h = "C/bass_fx.h";
      version = "2.4.12.1";
      so = {
        i686_linux = "libs/x86/libbass_fx.so";
        x86_64-linux = "libs/x86_64/libbass_fx.so";
      };
      urlpath = "z/0/bass_fx24-linux.zip";
      hash = "sha256-Hul2ELwnaDV8TDRMDXoFisle31GATDkf3PdkR2K9QTs=";
```
You can add yourself as a maintainer if you like.

By the way, your commits are not linked to your GH user. See [here](https://docs.github.com/en/pull-requests/committing-changes-to-your-project/troubleshooting-commits/why-are-my-commits-linked-to-the-wrong-user#commits-are-not-linked-to-any-user).

Comment (jacekpoz) [reply pkgs/development/libraries/audio/libbass/default.nix:20]: sorry for the long wait, had a few things to do

I also updated libbass_fx and added libbassmix, should I create a new pull request and bundle all of that there?

my commits being unverified and adding myself as a maintainer are postponed until I figure out how to selfhost an email server or finally decide to temporarily buy some service; is it alright if I set my email to a non-existent one and leave my commits unverified for now?

Comment (FliegendeWurst) [reply pkgs/development/libraries/audio/libbass/default.nix:20]: > is it alright if I set my email to a non-existent one and leave my commits unverified for now?

Unverified commits are fine, but we usually want a working email address in the maintainers list.

**Comment by theRoboxx** [comment]
Hi there, I'm new to the NixOS Eco System.
What is needed for this PR to be merged?

@jacekpoz What's your status?

**Comment by jacekpoz** [comment]
hi @theRoboxx, thanks for reminding me about this

I had another go at selfhosting mail, remembered I have a protonmail that's good enough for now and the only thing really needed right now is to update that hash which I'm on right now

**Comment by jacekpoz** [comment]
there we go, that should be all! I'll try to get back to the main reason I updated this library soon, expect more PRs on this file

**Comment by jacekpoz** [comment]
(repushed to get verified commits, finally figured that out)

**Comment by jacekpoz** [comment]
(repushed again to add my matrix id, should be the last one)

**Comment by sarahec** [comment]
Result of `nixpkgs-review pr 256414` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>libbass</li>
    <li>libbass_fx</li>
    <li>ultrastar-creator</li>
    <li>ultrastar-manager</li>
  </ul>
</details>

**Review by fpletz — APPROVED** [review]

**Pull request merged**